### DISCUSSION TITLE
Running in a Raspberry Pi Zero

### DISCUSSION THREAD

**Discussion by JuanM04** [dani-garcia/vaultwarden#1283] [Q&A] [answered]
Hi, I'm having some issues trying to build and run bitwarden_rs in a Raspberry Pi Zero.

First, I tried with docker, but it kept pulling the wrong image (even if I pulled armv6), and yes, I read #412.

Then, I tried cross-compiling, because I knew it would take forever to compile with the Pi CPU. After following every tutorial, `pkg-config` didn't let me finish, saying that "it doesn't support cross-compiling".

As a last resource, I tried compiling directly from the Pi, and it took like 8 hours to compile all the dependencies, but it takes so long to compile `bitwarden_rs` that the SSH connection closes.

If there is anyone that successfully built and/or run bitwarden_rs inside a Raspberry Pi Zero, how did you do it?

**Comment by BlackDex** [comment, marked as answer]
Have you tried running docker with the `--platform linux/arm/v6` option on the Pi it self with the docker command?

You can also try to use the `docker/arm32v6/Dockerfile` to build it on bigger hardware and extract the bitwarden_rs binary from it, and just copy that over to your Pi it self. That together with the web-vault from `https://github.com/dani-garcia/bw_web_builds/releases` should work just fine without docker running (Depending on which OS you have installed on the Pi. But if it is based upon the latest Debian, it should be fine.

Also, if  you want to compile it manually on the Pi it self, which indeed takes a very long time, you probably should run that command within a tmux or screen so it survives the session timeout. But you probably also need a large enough swap-file for all the memory it will consume.

**Reply by JuanM04** [reply]
Thank you, I didn't notice the Dockerfile. Below there are the commands I used for future people:

```bash
docker build -f docker/arm32v6/Dockerfile -t bitwarden_rs:pi-zero .
docker run -d --name bw bitwarden_rs:pi-zero
docker cp bw:/bitwarden_rs .
docker rm -f bw
```
